source/Fleet.cpp

```diff
@@ -205,19 +206,38 @@ void Fleet::Load(const DataNode &node)
 				variants.clear();
 				total = 0;
 			}
-			variants.emplace_back(child);
-			total += variants.back().weight;
+
+			int weight = 1;
+			string name;
+			bool named = false;
+			if(node.Size() >= 2 + add && !node.IsNumber(1 + add))
+			{
+				name = node.Token(1 + add);
+				named = true;
+			}
+			if(node.Size() >= 2 + add + named)
+				weight = child.Value(1 + add + named);
+
+			if(named)
+				variants.emplace_back(make_pair(GameData::Variants().Get(name), weight));
```

Comment:
Use the argument forwarding feature of `emplace_____` to select the appropriate constructor and create the emplaced type in-place